WIP(many): change color system

HerrTopi opened the pull request.

matyasf commented on packages/shared-types/src/Colors.ts:
```diff
@@ -22,121 +22,106 @@
-  backgroundDanger: BaseColors['crimson']
-  backgroundWarning: BaseColors['fire']
+type Contrasts = {
+  white: Primitives[keyof Primitives]
+  grey11: Primitives[keyof Primitives]
+  grey1214: Primitives[keyof Primitives]
+  grey1424: Primitives[keyof Primitives]
+  grey24: Primitives[keyof Primitives]
+  grey4570: Primitives[keyof Primitives]
+  grey5782: Primitives[keyof Primitives]
+  grey100: Primitives[keyof Primitives]
+  grey125: Primitives[keyof Primitives]
+
+  blue12: Primitives[keyof Primitives]
+  blue4570: Primitives[keyof Primitives]
+  blue5782: Primitives[keyof Primitives]
+
+  green12: Primitives[keyof Primitives]
+  green4570: Primitives[keyof Primitives]
+  green5782: Primitives[keyof Primitives]
+
+  orange12: Primitives[keyof Primitives]
+  orange3045: Primitives[keyof Primitives]
+  orange4570: Primitives[keyof Primitives]
+  orange5782: Primitives[keyof Primitives]
+
+  red12: Primitives[keyof Primitives]
+  red4570: Primitives[keyof Primitives]
+  red5782: Primitives[keyof Primitives]
```
what was the intention here? the types just resolve to `string`

HerrTopi replied:
There is value in expressing more than just strings (as we agreed on with shared-types/src/ComponentThemeVariables.ts below). The correct way to type them for examle:

grey1214: Primitives['grey12'] | Primitives['grey14']

matyasf commented on packages/shared-types/src/ComponentThemeVariables.ts:
```diff
@@ -40,8 +39,8 @@ export interface ComponentTheme {
 export type AlertTheme = {
-  background: Colors['backgroundLightest']
-  color: Colors['textDarkest']
+  background: string
+  color: string
```
Could we type these more exactly like in the old version?

HerrTopi replied:
Note: after deliberation, the team decided to go with the "old version", so I did a refactor

matyasf commented on packages/ui-theme-tokens/src/common/colors/pirimitves.ts:
typo: `pirimitives`

HerrTopi replied:
Thanks, fixed!

matyasf commented on packages/ui-theme-tokens/src/instructure/colors.ts:
```diff
@@ -23,9 +23,9 @@
+import { Colors } from '@instructure/shared-types'
 
-const values: BaseColors = {
+const values: any = {
```
no `any` please

HerrTopi replied:
I just left it in to make ts pass, I knew we will delete the whole instructure theme folder anyways

joyenjoyer commented on .github/workflows/pr-validation.yml:
```diff
@@ -18,8 +18,8 @@ jobs:
         run: npm run bootstrap
-      - name: Run test for the patchset
-        run: USE_REACT_STRICT_MODE=1 npm run test:patchset
+#      - name: Run test for the patchset
```
The full comment block can be deleted imo. Git preserves it anyways.

balzss replied:
I think this comment is temporary to get a preview build. should be put back

HerrTopi replied:
Indeed, this was here only for testing, already removed

joyenjoyer commented on docs/guides/themes.md:
```diff
@@ -0,0 +1,85 @@
+
+`UI` colors refer to the curated list of color-tokens that our users should be using when coloring the UI. Ideally, this list should contain every color, that an `Instructure` product will ever need. In rare exceptions, `contrasts` can be used if the design requires it.
+
+`UI` colors are built from `contrats`.
```
typo: contrats

HerrTopi replied:
Thanks, fixed!

HerrTopi committed:
feat(many): replace color system

BREAKING CHANGE:
Colors have changed, theme overrides might not work the same way as before
